nemo/collections/nlp/data/token_classification/token_classification_dataset.py

```diff
@@ -266,12 +267,21 @@ def __init__(
                 ignore_start_end=ignore_start_end,
             )
 
-            pickle.dump(features, open(features_pkl, "wb"))
+            # save features to a temp file first to make sure that non-master processes don't start reading the file
```

Comment:
This dosnt have any rank check - very easy to create hundreds of temp dirs and dump loads of files into them. Whoever is author, please rethink doing this.

Reply:
@ekmb or @PeganovAnton ^